core/src/ledger_cleanup_service.rs

```diff
@@ -107,12 +107,13 @@ impl LedgerCleanupService {
             if i == 0 {
                 debug!("purge: searching from slot: {}", slot);
             }
-            // Not exact since non-full slots will have holes
-            total_shreds += meta.received;
-            total_slots.push((slot, meta.received));
+            // Unrooted slots are not eligible for cleaning
             if slot > root {
                 break;
             }
+            // This method not exact since non-full slots will have holes
+            total_shreds += meta.received;
+            total_slots.push((slot, meta.received));
```

Comment:
This is the minor bug fix I mentioned - add a slot to clean candidates (`total_slots`) after the `slot > root` check. This doesn't affect validators with real limit_ledger_size values since a potential slot that is newer than `root` would be the newest slot in `total_slots`, and as such, this value wouldn't get picked for cleaning in the bottom half of this function.

However, it was making my unit test fail and the new order of operations is the more proper thing to do.